lib/my_app/customer/customer.ex

```diff
@@ -58,6 +58,43 @@ defmodule MyApp.Customer do
     |> Repo.insert()
   end
 
+  @spec create_many_profiles([map]) :: {:ok, [Profile.t()]} | {:error, [Profile.changeset()]}
+  @doc """
+  Creates several profiles at once.
+
+  This operation is atomic.
+
+  On error, only the invalid changesets are returned.
+
+  ## Examples
+
+      iex> create_many_profiles([
+      ...>   %{name: "Roberval Santos", age: 32},
+      ...>   %{name: "Levandovski Dmitri de Jesus", age: 41}
+      ...> ])
+      {:ok, [%Profile{}, %Profile{}]}
+
+      iex> create_many_profiles([
+      ...>   %{name: "Weeabito Brasil", age: 21},
+      ...>   %{name: "Invalidilson", age: -21}
+      ...> ])
+      {:error, [%Ecto.Changeset{}]}
+  """
+  def create_many_profiles(input_list) do
+    Repo.transaction(fn ->
+      results =
+        input_list
+        |> Enum.map(&create_profile/1)
```

Comment:
This will hit on DB for each entry purposefully. This is to ensure that we have _all_ invalid entries at once (shall the operation fail). This is not a problem because the profile PK is an GUID and thus does not depend on a DB sequence.